db/database.go

```diff
@@ -393,30 +404,43 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 				provider.CallbackURL = &updatedCallback
 			}
 
-			context.OIDCProviders[name] = provider
+			dbContext.OIDCProviders[name] = provider
 		}
-		if len(context.OIDCProviders) == 0 {
+		if len(dbContext.OIDCProviders) == 0 {
 			return nil, errors.New("OpenID Connect defined in config, but no valid OpenID Connect providers specified.")
 		}
 
 	}
 
 	// watchDocChanges is used for bucket shadowing
-	if !context.UseXattrs() {
-		go context.watchDocChanges()
+	if !dbContext.UseXattrs() {
+		go dbContext.watchDocChanges()
 	} else {
 		// Set the purge interval for tombstone compaction
-		context.PurgeInterval = DefaultPurgeInterval
+		dbContext.PurgeInterval = DefaultPurgeInterval
 		gocbBucket, ok := base.AsGoCBBucket(bucket)
 		if ok {
 			serverPurgeInterval, err := gocbBucket.GetMetadataPurgeInterval()
 			if err != nil {
 				base.Warnf(base.KeyAll, "Unable to retrieve server's metadata purge interval - will use default value. %s", err)
 			} else if serverPurgeInterval > 0 {
-				context.PurgeInterval = serverPurgeInterval
+				dbContext.PurgeInterval = serverPurgeInterval
+			}
+		}
+		base.Infof(base.KeyAll, "Using metadata purge interval of %.2f days for tombstone compaction.", float64(dbContext.PurgeInterval)/24)
+
+		if dbContext.Options.CompactInterval != 0 {
+			if autoImport {
+				db := Database{DatabaseContext: dbContext}
+				NewBackgroundTask("Compact database", dbContext.Name, func(ctx context.Context) error {
```

Comment:
Task name just as "Compact" would probably be clearer (and shorter)

```suggestion
				NewBackgroundTask("Compact", dbContext.Name, func(ctx context.Context) error {
```